clients/python/test/test_osparc/test_basic.py

```diff
@@ -1,6 +1,56 @@
+import json
+import subprocess
+from pathlib import Path
+from typing import Any, Dict, List, Set
+
 import osparc
 
+_PYTHON_DIR: Path = Path(__file__).parent.parent.parent
+
 
 def test_get_api():
     info = osparc.openapi()
     assert isinstance(info["info"]["version"], str)
+
+
+def test_dependencies(tmp_path: Path):
+    """
+    Ensure packages which are imported in osparc are also specified in setup.py
+    """
+    # get imported packages
+    import_file: Path = tmp_path / "imported_packages.txt"
```

Comment:
this is correct but a bit strange way to test dependencies. e.g. in `osparc-simcore` we guarantee dependencies because we use tools "cocktail" including pip-tools, mypy, pylint, etc

e.g. Why tests are not failing if there is an import missing?

Reply:
I have encountered 2 issues already with the dependencies:

1. I forgot to specify a requirement in `setup.py`. I believe I didn't discover this issue because the same dependency is required by the testing. Hence, simply doing `make install-test` already ensured that package was installed.
2. Apparently `osparc` requires a specific version of `tqdm`. This is harder to test as the requirements in `setup.py` should be as flexible as possible. So I don't really know how to automate that. 

This test is there to detect issue number 1. I checks that the packages I import within the source is actually specified in `setup.py`. I don't know of any other tool that really does exactly that.

Reply:
What I like about the `pipreqs` package is that it determines the dependencies by inspecting the source code which is exactly what I would like to do

Reply:
I guess the key differences in osparc-simcore repo are:
- we pin all dependencies
- we always install them in containers before shipping (i.e. never via pipy)

For that reason we do not need to do these checks

I still think that these tests (even if they are totally correct) are a bit over-engineered and I am convinced it fully guarantees that there will be no more errors in the future. I would rather test the code via workflows that start with `pip install osparc`.

Reply:
I definitely agree this is over-engineered. But that's because no package exists which does the correct check. That package should exist... I think this is the right check to make...